File: CS/CustomFilterControlPopupMenu/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraEditors
Imports System.Collections.Generic
Imports DevExpress.XtraEditors.Filtering

Namespace CustomFilterControlPopupMenu
	Partial Public Class Form1
		Inherits XtraForm
		Public Sub New()
			InitializeComponent()
			FilterControl.MenuManager = New CustomMenuManager(BarManager)
			FilterControl.UseMenuForOperandsAndOperators = True
		End Sub

		Private Sub OnPopupMenuShowing(ByVal sender As Object, ByVal e As PopupMenuShowingEventArgs) Handles FilterControl.PopupMenuShowing
			If e.MenuType = FilterControlMenuType.Column Then
				e.Menu.Tag = "custom"
			End If
		End Sub
	End Class
End Namespace

File: CS/CustomFilterControlPopupMenu/Form1.cs
using System;
using DevExpress.XtraEditors;
using System.Collections.Generic;
using DevExpress.XtraEditors.Filtering;

namespace CustomFilterControlPopupMenu {
    public partial class Form1 :XtraForm {
        public Form1() {
            InitializeComponent();
            FilterControl.MenuManager = new CustomMenuManager(BarManager);
            FilterControl.UseMenuForOperandsAndOperators = true;
        }

        private void OnPopupMenuShowing(object sender, PopupMenuShowingEventArgs e) {
            if (e.MenuType == FilterControlMenuType.Column)
                e.Menu.Tag = "custom";
        }
    }
}